shoplist/src/main/java/com/netguru/codereview/ui/MainFragment.kt

```diff
@@ -31,35 +27,54 @@ class MainFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
 
-        viewModel!!.shopLists.observe(this, { lists ->
-            val progressBar = view.findViewById<ProgressBar>(R.id.message)
-            val latestIcon = view.findViewById<ImageView>(R.id.latest_list_icon)
+        observe()
 
-            val shopLists = lists.map { mapShopList(it.first, it.second) }.also {
-                latestIcon?.load(it.first().iconUrl)
-            }
+        viewModel.getShopList()
+    }
+
+    private fun observe() {
+        viewModel.shopListsLiveData.observe(viewLifecycleOwner) { result ->
+            val progressBar = view?.findViewById<ProgressBar>(R.id.progress_bar)
+            val latestIcon = view?.findViewById<ImageView>(R.id.latest_list_icon)
+
+            result?.apply {
+                when(this) {
+                    is ResultState.Success -> {
+                        this.data?.map { shotList ->
+                            Log.i("LOG_TAG", "${
+                                shotList?.listName
+                            } : ${
+                                shotList?.items?.map {
+                                    it?.name
+                                }
+                            }"
+                            )
+                        }
+
+                        latestIcon?.load(this.data?.lastOrNull()?.iconUrl)
```

Comment:
Use state to handle errors, notify UI about the error if it happened, https://developer.android.com/kotlin/coroutines/coroutines-best-practices#exceptions,
this will be happened when having state of data,either success or error